android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -394,10 +394,10 @@ public static boolean executeHardwareKeystroke(int code, int shift, int lstates,
 
   public static boolean executeHardwareKeystroke(
     int code, int shift, KeyboardType keyboard, int lstates, int eventModifiers) {
-    if (keyboard == KeyboardType.KEYBOARD_TYPE_INAPP) {
+    if (keyboard == KeyboardType.KEYBOARD_TYPE_INAPP && InAppKeyboard != null) {
       InAppKeyboard.executeHardwareKeystroke(code, shift, lstates, eventModifiers);
       return true;
-    } else if (keyboard == KeyboardType.KEYBOARD_TYPE_SYSTEM) {
+    } else if (keyboard == KeyboardType.KEYBOARD_TYPE_SYSTEM && SystemKeyboard != null) {
       SystemKeyboard.executeHardwareKeystroke(code, shift, lstates, eventModifiers);
       return true;
     }
```

Comment:
Should we be logging an `else { KMLog.LogError(); }` here? Although if it happens on every keystroke that would kinda spam our logs, so perhaps it should happen just once per session.